### ISSUE TITLE
IonStruct object does not match instanceof dom.Value when used inside JSON stringify replacer

### ISSUE THREAD

**Issue by goldsail** [amazon-ion/ion-js#751]
Code: https://gist.github.com/goldsail/37a48b6ee4fb2bf363dac0012fa0263b

How to recreate: Run the script with NodeJS 18 (or any other NodeJS version) and latest version of ion-js (4.3.0).

I notice that IonStruct object does not match `instanceof dom.Value` when it's called inside a `JSON.stringify` replacer (line 23), unlike IonList and IonString. However, `instanceof dom.Value` will return true when I call it outside the replacer (line 14).

This seems a bug. Not sure if it's related to [this Proxy object conversion](https://github.com/amazon-ion/ion-js/blob/ff9741053918df9d918d4e667f62e1a8d570cc5b/src/dom/Struct.ts#L56) logic. But I definitely expect to see `instanceof dom.Value` returns true for an IonStruct inside the replacer, with consistent behavior of IonList and IonString.

---

I spotted this bug when I was trying to solve this use case: We have a JSON object, inside which we put ion-js objects as nested fields. Example:

```
{
  "id": "123",
  "name": "john",
  "data": <some_ion_object>
}
```

And we want to pretty print this object on the webpage. The first solution that comes to my mind is `JSON.stringify(object, null, 2)`. However, the ion type annotations will be ignored (it prints ion value in JSON format). To retain everything specific to Ion, we had to resort to `JSON.stringify(object, replacer, 2)` where

```
const replacer = (key, value) => {
  if (value instanceof dom.Value) {
    // print ion values differently. This will retain ion type annotations.
    return dumpPrettyText(value);
  }
  // otherwise, pass it to default JSON stringify behavior.
  return value;
};
```

**Comment by desaikd** [comment]
After investigating this I found out where the issue was:

JSON.stringify() uses toJSON() if the object has that defined as per [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON()_behavior):

> If the value has a toJSON() method, it's responsible to define what data will be serialized. Instead of the object being serialized, the value returned by the toJSON() method when called will be serialized.

#### Solution: 

- Previously the toJSON() for struct was returning an object based on null object prototype which was giving incorrect results for instanceof when called inside the JSON.stringify replacer function.
- Now we are using Struct's prototype to for the return object on toSJON().

**Comment by desaikd** [comment]
Fixed with #752
